src/iface/interface.rs

```diff
@@ -1082,24 +1082,16 @@ impl<'a> Interface<'a> {
             let mut respond = |inner: &mut InterfaceInner, response: IpPacket| {
                 neighbor_addr = Some(response.ip_repr().dst_addr());
                 match device.transmit().ok_or(Error::Exhausted) {
-                    Ok(_t) => {
+                    Ok(t) => {
                         #[cfg(feature = "proto-sixlowpan-fragmentation")]
-                        if let Err(e) = inner.dispatch_ip(_t, response, Some(_out_packets)) {
-                            net_debug!("failed to dispatch IP: {}", e);
-                            return Err(e);
-                        }
+                        inner.dispatch_ip(t, response, Some(_out_packets))?;
 
                         #[cfg(not(feature = "proto-sixlowpan-fragmentation"))]
-                        if let Err(e) = inner.dispatch_ip(_t, response, None) {
-                            net_debug!("failed to dispatch IP: {}", e);
-                            return Err(e);
-                        }
+                        inner.dispatch_ip(t, response, None)?;
+
                         emitted_any = true;
                     }
-                    Err(e) => {
-                        net_debug!("failed to transmit IP: {}", e);
-                        return Err(e);
-                    }
+                    Err(e) => return Err(e),
```

Comment:
The entire match expression can be dropped if the error is just being passed through.

Reply:
I removed the match expression now.